internals/servicelog/ringbuffer.go

```diff
@@ -258,6 +272,8 @@ func (rb *RingBuffer) TailIterator() Iterator {
 // backwards from the head.
 func (rb *RingBuffer) HeadIterator(lines int) Iterator {
 	firstLine := rb.reverseLinePosition(lines)
+	rb.rwlock.RLock()
+	defer rb.rwlock.RUnlock()
```

Comment:
This is now a bit messy, as `rwlock` is locked and unlocked twice now, once inside `reverseLinePosition` and once here. Could we get rid of the locking in `reverseLinePosition` and do it once in this function?